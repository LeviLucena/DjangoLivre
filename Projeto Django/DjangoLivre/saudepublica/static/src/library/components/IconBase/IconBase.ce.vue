<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
