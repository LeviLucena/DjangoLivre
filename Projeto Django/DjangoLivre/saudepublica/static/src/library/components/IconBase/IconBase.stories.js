import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
