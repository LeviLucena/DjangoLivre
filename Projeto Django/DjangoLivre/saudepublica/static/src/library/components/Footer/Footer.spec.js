import { shallowMount } from '@vue/test-utils'
import IconBase from '../IconBase/IconBase.ce.vue'
import BrList from '../List/List.ce.vue'
import BrFooter from './Footer.ce.vue'

describe('Footer', () => {
  const propLogo = `{url: 'https://url-da-logo', description: 'descricao-da-logo'}`
  const propCategorias = `[
    {
      title: 'Categoria 1',
      items: [
        {title: 'item 1', href: '/item1'},
        {title: 'item 2', href: '/item1'},
        {title: 'item 3', href: '/item1'},
        {title: 'item 4', href: '/item1'}
      ]
    },
    {
      title: 'Categoria 2',
      items: [
        {title: 'item 5', href: 'http://www.google.com'},
        {title: 'item 6', href: '/item1'},
        {title: 'item 7', href: '/item1'},
        {title: 'item 8', href: '/item1'}
      ]
    }
  ]`
  const propSocial = `{
    label: 'Título',
    networks: [
      {
        href: '#',
        image: 'https://url-da-rede-social',
        description: 'descricao-da-rede-social'
      },
      {
        href: '#',
        image: 'https://url-da-rede-social',
        description: 'descricao-da-rede-social'
      },
      {
        'href': '#',
        'icon': 'facebook-f',
        'iconFamily': 'fab',
        'description': 'Facebook'
      },
      {
        'href': '#',
        'icon': 'twitter',
        'iconFamily': 'fab',
        'description': 'Twitter'
      }
    ]
  }`
  const propPartners = `[
    {
      href: '#',
      image: 'https://url-do-parceiro',
      description: 'descricao-do-parceiro'
    },
    {
      href: '#',
      image: 'https://url-do-parceiro',
      description: 'descricao-do-parceiro'
    }
  ]`

  const textHello = 'hello'
  const wrapperFooter = shallowMount(BrFooter, {
    propsData: {
      text: textHello,
      inverted: false,
      categories: propCategorias,
      logo: propLogo,
      social: propSocial,
      partners: propPartners,
    },
    global: {
      stubs: {
        'br-list': BrList,
        'icon-base': IconBase,
      },
    },
  })

  const wrapperFooterInverted = shallowMount(BrFooter, {
    propsData: {
      logo: propLogo,
      text: textHello,
      inverted: true,
      categories: propCategorias,
    },
  })

  test('it renders footer text', () => {
    expect(wrapperFooter.text()).toMatch(textHello)
  })

  test('it renders inverted mode', () => {
    expect(wrapperFooterInverted.find('.inverted').exists()).toBe(true)
  })

  test('it renders categories', () => {
    expect(wrapperFooter.findAll('.br-list')).toHaveLength(3)
    expect(wrapperFooter.findAll('.br-item')).toHaveLength(10)
  })

  test('it opens and close category list when click on it', () => {
    global.innerWidth = 200
    global.dispatchEvent(new Event('resize'))
    wrapperFooter
      .find('a')
      .trigger('click')
      .then(() => {
        expect(wrapperFooter.find('.open').exists()).toBe(true)
        wrapperFooter
          .find('a')
          .trigger('click')
          .then(() => {
            expect(wrapperFooter.find('.open').exists()).toBe(false)
          })
      })
  })

  test('it renders logo', () => {
    expect(wrapperFooter.find('.logo').exists()).toBe(true)
  })

  test('it renders logo image', () => {
    expect(wrapperFooter.find(`img[src="https://url-da-logo"]`).exists()).toBe(true)
  })

  test('it renders social networks', () => {
    expect(wrapperFooter.find('.social-network').exists()).toBe(true)
    expect(wrapperFooter.element).toMatchSnapshot()
  })

  test('it renders social networks label', () => {
    expect(wrapperFooter.text()).toMatch('Título')
  })

  test('it renders social networks itens', () => {
    expect(wrapperFooter.find(`img[src="https://url-da-rede-social"]`).exists()).toBe(true)
  })

  test('it renders footer images', () => {
    expect(wrapperFooter.find(`img[src="https://url-do-parceiro"]`).exists()).toBe(true)
  })
})

describe('Footer Basico', () => {
  const wrapperFooter = shallowMount(BrFooter, {
    propsData: {
      logo: 'propLogo',
    },
  })
  const wrapperFooterInverted = shallowMount(BrFooter, {
    propsData: {
      logo: 'propLogo',
      inverted: true,
    },
  })

  test('it renders footer text', () => {
    expect(wrapperFooter.find('.info').exists()).toBe(false)
  })

  test('it renders inverted mode', () => {
    expect(wrapperFooterInverted.find('.inverted').exists()).toBe(true)
  })

  test('it renders categories', () => {
    expect(wrapperFooter.find('.br-list').exists()).toBe(false)
  })

  test('it renders logo', () => {
    expect(wrapperFooter.find('.logo').exists()).toBe(true)
  })

  test('it renders logo image', () => {
    expect(wrapperFooter.find(`img[src]`).exists()).toBe(false)
  })

  test('it renders social networks', () => {
    expect(wrapperFooter.find('.social-network').exists()).toBe(false)
  })

  test('it renders social networks itens', () => {
    expect(wrapperFooter.find(`img[src="https://url-da-rede-social"]`).exists()).toBe(false)
  })

  test('it renders footer images', () => {
    expect(wrapperFooter.find(`img[src="https://url-do-parceiro"]`).exists()).toBe(false)
  })
})
