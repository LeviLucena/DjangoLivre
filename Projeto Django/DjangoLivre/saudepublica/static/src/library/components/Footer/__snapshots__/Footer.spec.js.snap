// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer it renders social networks 1`] = `
<footer
  class="br-footer"
>
  <div
    class="container-lg"
  >
    <div
      class="logo"
    >
      <img
        alt="descricao-da-logo"
        src="https://url-da-logo"
      />
    </div>
    <div
      class="br-list horizontal"
      data-toggle="true"
      data-unique="true"
      density="large"
      role="list"
    >
      <!--v-if-->
      <!--v-if-->
      
      
      <div
        class="col-2"
      >
        <a
          class="br-item header"
          href="javascript:void(0)"
        >
          <div
            class="content text-down-01 text-bold text-uppercase"
          >
            Categoria 1
          </div>
          <!--v-if-->
        </a>
        <div
          class="br-list"
        >
          <span
            class="br-divider d-md-none"
          />
          
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 1
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 2
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 3
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 4
            </div>
          </a>
          
          <span
            class="br-divider d-md-none"
          />
        </div>
      </div>
      <div
        class="col-2"
      >
        <a
          class="br-item header"
          href="javascript:void(0)"
        >
          <div
            class="content text-down-01 text-bold text-uppercase"
          >
            Categoria 2
          </div>
          <!--v-if-->
        </a>
        <div
          class="br-list"
        >
          <span
            class="br-divider d-md-none"
          />
          
          <a
            class="br-item"
            href="http://www.google.com"
          >
            <div
              class="content"
            >
              item 5
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 6
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 7
            </div>
          </a>
          <a
            class="br-item"
            href="/item1"
          >
            <div
              class="content"
            >
              item 8
            </div>
          </a>
          
          <span
            class="br-divider d-md-none"
          />
        </div>
      </div>
      
      
    </div>
    <div
      class="d-none d-sm-block"
    >
      <div
        class="row align-items-end justify-content-between py-5"
      >
        <div
          class="col social-network"
        >
          <div
            class="social-network-title"
          >
            TÃ­tulo
          </div>
          <div
            class="d-flex"
          >
            
            <a
              class="mr-3"
              href="#"
              title="descricao-da-rede-social"
            >
              <img
                alt="descricao-da-rede-social"
                src="https://url-da-rede-social"
              />
              <!--v-if-->
            </a>
            <a
              class="mr-3"
              href="#"
              title="descricao-da-rede-social"
            >
              <img
                alt="descricao-da-rede-social"
                src="https://url-da-rede-social"
              />
              <!--v-if-->
            </a>
            <a
              class="br-button circle"
              href="#"
              title="Facebook"
            >
              <!--v-if-->
              <i
                aria-hidden="true"
                class="fab fa-facebook-f"
              />
            </a>
            <a
              class="br-button circle"
              href="#"
              title="Twitter"
            >
              <!--v-if-->
              <i
                aria-hidden="true"
                class="fab fa-twitter"
              />
            </a>
            
          </div>
        </div>
        <div
          class="col assigns text-right"
        >
          
          <a
            href="#"
          >
            <img
              alt="descricao-do-parceiro"
              class="ml-4 logo-footer-ajuste"
              src="https://url-do-parceiro"
            />
          </a>
          <a
            href="#"
          >
            <img
              alt="descricao-do-parceiro"
              class="ml-4 logo-footer-ajuste"
              src="https://url-do-parceiro"
            />
          </a>
          
        </div>
      </div>
    </div>
  </div>
  <span
    class="br-divider my-3"
  />
  <div
    class="container-lg"
  >
    <div
      class="info"
    >
      <div
        class="text-down-01 text-extra-bold pb-3 text-center"
      >
        hello
      </div>
    </div>
  </div>
</footer>
`;
